### DISCUSSION TITLE
Enhancing descriptions for the usage of scripted data sources

### DISCUSSION THREAD

**Discussion by Markus-Elfring** [eclipse-birt/birt#1014] [Q&A]
Some information was published also for [the usage of scripted data sources](https://www.linuxtopia.org/online_books/eclipse_documentation/eclipse_birt_report_guide/topic/org.eclipse.birt.doc/birt/eclipse_birt_report_connecting.2.1.html "Connecting to a data source") by the “Business Intelligence Reporting Tool”.
:thought_balloon: I would appreciate if corresponding descriptions can be updated and improved further.
Would you like to influence the situation any more for available software documentation? :thinking:

**Comment by hvbtup** [comment]
I attached a simple example which can act like a "for-loop".

Basically, you need to add some code for initialization (in the "open" event) and possibly cleanup ("close" event) and to detect if another record should be returned and if so, set its columns accordingly (in the "fetch" event).

[scripted_ds_example.zip](https://github.com/eclipse/birt/files/9157255/scripted_ds_example.zip)

**Reply by Markus-Elfring** [reply]
:thought_balloon: Can the documentation become clearer accordingly?

Will any additional methods become also relevant for the safer usage of this programming interface? :thinking:
